src/app/parcels/fetchParcelTableData.ts

```diff
@@ -30,52 +35,158 @@ export const getCongestionChargeDetailsForParcels = async (
 export type ParcelProcessingData = Awaited<ReturnType<typeof getParcelProcessingData>>;
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export const getParcelProcessingData = async (supabase: Supabase, dateRange: DateRangeState) => {
-    const { data, error } = await supabase
-        .from("parcels")
-        .select(
-            `
-        parcel_id:primary_key,
-        
-        collection_centre:collection_centres ( 
-            name, 
-            acronym
-         ),
-         
-        collection_datetime,
-        packing_datetime,
-        voucher_number,
-        
-        client:clients (
-            primary_key,
-            full_name,
-            address_postcode,
-            flagged_for_attention,
-            signposting_call_required,
-            phone_number,
-            
-            family:families (
-                age,
-                gender
-            )
-        ),
-        
-        events (
-            event_name,
-            event_data,
-            timestamp
-        )
-    `
-        )
-        .gte("packing_datetime", dateRange.from)
-        .lte("packing_datetime", dateRange.to)
-        .order("packing_datetime", { ascending: false })
-        .order("timestamp", { ascending: false, foreignTable: "events" })
-        .limit(1, { foreignTable: "events" });
+export const getParcelProcessingData = async (
+    supabase: Supabase,
+    filters: Filter<ParcelsTableRow, any>[],
+    sortState: SortState<ParcelsTableRow>,
+    start?: number,
+    end?: number,
+    parcelIds?: string[]
+) => {
+    let query = supabase.from("parcels_plus").select("*");
+
+    filters.forEach((filter) => {
+        if (filter.methodConfig.methodType === PaginationType.Server) {
+            query = filter.methodConfig.method(query, filter.state);
+        }
+    });
+
+    if (sortState.sort && sortState.column.sortMethodConfig?.methodType === PaginationType.Server) {
+        query = sortState.column.sortMethodConfig.method(query, sortState.sortDirection);
+    } else {
+        query = query.order("packing_datetime", { ascending: false });
+    }
```

Comment:
this is the default sorting behaviour for the parcels table that is shown on initial render, when sort state is set to {sort: false}. see my review comments on ParcelsPage.tsx